### PR TITLE
decimals for revival

### PR DESCRIPTION
**Pull request by Ledespana**

### CODE REVIEW

**Review by tehnalogos — DISMISSED** [review]

File: frontend/components/LSP7Panel.tsx
Comment (tehnalogos) [inline frontend/components/LSP7Panel.tsx:1]: @Ledespana in this file we should take the raw value that we stored for balance and convert it to a displayBalance constant that we can then show to the user

File: frontend/components/LSP7Panel.tsx [hunk 1]
```diff
@@ -93,10 +94,13 @@ const LSP7Panel: React.FC<LSP7PanelProps> = ({
...
       );
       const lsp7 = tokenContract.connect(signer);
+    
+      const balanceInBaseUnits = ethers.utils.parseUnits(tokenData?.balance as string, tokenData?.decimals);
+
       const lsp7Tx = lsp7.interface.encodeFunctionData('transfer', [
         vaultAddress,
         await signer.getAddress(),
-        tokenData?.balance,
+        balanceInBaseUnits,
```
Comment (tehnalogos) [inline frontend/components/LSP7Panel.tsx:103]: this could be wrong value if you dont convert data when we first get it

Comment (Ledespana) [reply frontend/components/LSP7Panel.tsx:103]: the conversion is in line 98

File: frontend/components/LSP7Panel.tsx [hunk 2]
```diff
@@ -33,6 +33,16 @@ const LSP7Panel: React.FC<LSP7PanelProps> = ({
...
 }) => {
+  const displayTokenAmount = tokenData?.balance && tokenData?.decimals
+    ? parseFloat(ethers.utils.formatUnits(tokenData?.balance, tokenData?.decimals)).toFixed(
+      tokenData.tokenType === LSP4_TOKEN_TYPES.TOKEN ? Number(tokenData?.decimals) : 0
+      )
+    : '0';
+
+  // Assuming rawTokenAmount is a BigNumber representing the amount in base units
+  const rawTokenAmount = tokenData?.balance
+  console.log('rawTokenAmount', rawTokenAmount)
```
Comment (democrazyfi) [inline frontend/components/LSP7Panel.tsx:44]: we need this console log and the one below?

File: frontend/utils/tokenUtils.ts [hunk 3]
```diff
@@ -138,17 +138,18 @@ export const getLSPAssetBasicInfo = async (
...
-        ? parseFloat(ethers.utils.formatUnits(_balance, decimals)).toFixed(
-            LSP4TokenType === LSP4_TOKEN_TYPES.TOKEN ? 4 : 0
-          )
-        : '0';
+
+      // balance = _balance
+      //   ? parseFloat(ethers.utils.formatUnits(_balance, decimals)).toFixed(
+      //       LSP4TokenType === LSP4_TOKEN_TYPES.TOKEN ? decimals : 0
+      //     )
+      //   : '0';
```
Comment (democrazyfi) [inline frontend/utils/tokenUtils.ts:152]: not needed?

**Comment by democrazyfi** [comment]
Are we expected not to see balance here? 

https://deploy-preview-83--universal-grave.netlify.app/grave/0x7DdEE8C820536c75cD0b47a92de22Df75C131838

Screenshot 2024-02-25 at 17 04 01

**Comment by Ledespana** [comment]
> Are we expected not to see balance here?
> 
> https://deploy-preview-83--universal-grave.netlify.app/grave/0x7DdEE8C820536c75cD0b47a92de22Df75C131838
> 
> Screenshot 2024-02-25 at 17 04 01

ohhh I've just update my branch and the lastest update in main causes this

**Review by democrazyfi — DISMISSED** [review]

File: frontend/components/LSP7Panel.tsx [hunk 4]
```diff
@@ -33,6 +33,15 @@ const LSP7Panel: React.FC<LSP7PanelProps> = ({
   vaultOwner,
   onReviveSuccess,
 }) => {
+  const displayTokenAmount = tokenData?.balance !== undefined && tokenData?.decimals !== undefined
```
Comment (tehnalogos) [inline frontend/components/LSP7Panel.tsx:36]: nested ifs makes this a bit difficult to follow

File: frontend/components/LSP7Panel.tsx [hunk 5]
```diff
@@ -95,10 +105,18 @@ const LSP7Panel: React.FC<LSP7PanelProps> = ({
         signer
       );
       const lsp7 = tokenContract.connect(signer);
+      let balanceInBaseUnits = '0'
+      if (tokenData && tokenData.decimals !== undefined && rawTokenAmount) {
+        balanceInBaseUnits = ethers.utils.formatUnits(rawTokenAmount, tokenData?.decimals);
```
Comment (tehnalogos) [inline frontend/components/LSP7Panel.tsx:110]: if we never touch the raw value that we receive in tokenUtils as the balance is this step necessary?

File: frontend/components/LSP7Panel.tsx [hunk 6]
```diff
@@ -33,6 +33,15 @@ const LSP7Panel: React.FC<LSP7PanelProps> = ({
...
   onReviveSuccess,
 }) => {
+  const displayTokenAmount = tokenData?.balance !== undefined && tokenData?.decimals !== undefined
+    ? parseFloat(ethers.utils.formatUnits(tokenData?.balance, tokenData?.decimals)).toFixed(
+      tokenData.tokenType === LSP4_TOKEN_TYPES.TOKEN ? Number(tokenData?.decimals) : 0
+      )
+    : '0';
+
+  // Assuming rawTokenAmount is a BigNumber representing the amount in base units
+  const rawTokenAmount = tokenData?.balance
```
Comment (tehnalogos) [inline frontend/components/LSP7Panel.tsx:43]: Screenshot 2024-02-26 at 7 39 18 AM
This is quite difficult to follow. I think worth implementing a function that dynamically changes how many decimals to display. Chat GPT would good at this

**Review by tehnalogos — APPROVED** [review]

**Pull request merged**